bndtools.m2e/src/bndtools/m2e/IndexConfigurator.java

```diff
@@ -108,6 +108,91 @@ public WorkspaceRepository getRepository() {
         }
     }
 
+    private final class RebuildIndexCheck extends WorkspaceJob {
+
+        private final List<IResourceChangeEvent> events = new ArrayList<>();
+        private final IMavenProjectFacade facade;
+
+        private boolean noMoreEvents;
+
+        public RebuildIndexCheck(String name, IResourceChangeEvent event, IMavenProjectFacade facade) {
+            super(name);
+            this.events.add(event);
+            this.facade = facade;
+        }
+
+        public void addEvent(IResourceChangeEvent event) {
+            synchronized (pendingJobs) {
+                if (!noMoreEvents) {
```

Comment:
The `noMoreEvent` boolean is not necessary. `addEvent` is only called under `pendingJob` lock when then job is in the map.

Reply:
I'd quite like to leave this in as it makes it much harder to accidentally use the Job incorrectly. Events really can't be added after a certain point in the RebuildIndexCheck lifecycle, and if someone tries to add one later then they really should get an exception. The other option is to add lots of JavaDoc explaining that you always have to synchronize and check the pendingJobs, but that seems like an accident waiting to happen...